src/VisualStudio/Core/Def/Experimentation/EnhancedColorExperiment.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using EnvDTE;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Settings;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.Win32;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.Experimentation
+{
+    [Export(typeof(IWpfTextViewConnectionListener))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Analyzable)]
+    internal class EnhancedColorExperiment : ForegroundThreadAffinitizedObject, IWpfTextViewConnectionListener
+    {
+        private const string UseEnhancedColorsFlight = "UseEnhancedColors";
+        private const string StopEnhancedColorsFlight = "StopEnhancedColors";
+        private const string UseEnhancedColorsSetting = "WindowManagement.Options.UseEnhancedColorsForManagedLanguages";
+
+        private readonly IExperimentationService _experimentationService;
+        private readonly IServiceProvider _serviceProvider;
+
+        private EnhancedColorApplier _colorApplier;
+        private ISettingsManager _settingsManager;
+        private RegistryKey _vsRegistryRoot;
+
+        private bool _inUseEnhancedColorsFlight;
+        private bool _inStopEnhancedColorsFlight;
+        private bool _hasTextViewOpened;
+
+        [ImportingConstructor]
+        private EnhancedColorExperiment(IThreadingContext threadingContext, [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, VisualStudioExperimentationService experimentationService)
+            : base(threadingContext)
+        {
+            _serviceProvider = serviceProvider;
+            _experimentationService = experimentationService;
+        }
+
+        public void SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers)
+        {
+            AssertIsForeground();
+
+            if (!_hasTextViewOpened)
+            {
+                _hasTextViewOpened = true;
+
+                _colorApplier = new EnhancedColorApplier(_serviceProvider);
+                _vsRegistryRoot = VSRegistry.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_UserSettings, writable: true);
```

Comment:
Where do you read the value of `_vsRegistryRoot`?

Reply:
Thanks, I should have removed it when I changed to using the SettingsManager for getting the current theme id.